modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFileVariation.pm

```diff
@@ -0,0 +1,183 @@
+  my $species = $self->param('species');
+  my $current_step = $self->param('current_step') ;
+  my $output = $self->param('output_path');
+  my $app_path = $self->param('app_path'); 
+  my $genome_data = {
+    dbname     => $self->param('dbname'),
+    gca        => $self->param('gca'),
+    genome_id  => $self->param('genome_id'),
+    species    => $self->param('species'),
+    version    => $self->param('version'),
+    type       => $self->param('type'),  
+    root_path  => path($self->param('root_path'))
+  };
+  my $dba = Bio::EnsEMBL::Registry->get_DBAdaptor( $self->param('species'), $self->param('group') );
+  my $lookup = Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup->new("genome_data" => $genome_data); 
+  my $genome = $lookup->get_genome('1');
+  my $variation_bed_writer = Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::variants::VariationBedWriter->new(root_path => path($self->param('root_path')));
+  ##my $release = get_release_versions();
+
+  $self->clear_output_folder($variation_bed_writer, $genome);
+  my $vcf_files = $self->get_vcf_files($genome);
+
+  foreach my $vcf_file_path (@{$vcf_files}) {
+    $self->warning($vcf_file_path);
+    $self->write_bed_from_vcf($vcf_file_path, $variation_bed_writer, $genome);
+  }
+  
+  $self->scale_bed_files($variation_bed_writer, $genome);
+  $self->generate_bigbeds($variation_bed_writer, $genome)
+}
+
+
+sub scale_bed_files {
+  my ($self, $variation_bed_writer, $genome) = @_;
+  foreach my $bed_file_path ($variation_bed_writer->variants_path($genome->genome_id())->children) {
+    Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::variants::VariantsScaler->new(
+                                     input_bed_file => $bed_file_path, 
+                                     genome_id => $genome->genome_id(), genome => $genome, 
+                                     variant_path => $variation_bed_writer->variants_path($genome->genome_id()) )->scale_variants_bed();
+    $bed_file_path->remove;
+  }
+}
+
+
+sub clear_output_folder {
+  my ($self, $variation_bed_writer, $genome ) = @_;
+  foreach my $file_path ($variation_bed_writer->variants_path($genome->genome_id())->children) {
+    $file_path->remove;
+  };
+}
+
+
+sub generate_bigbeds {
+  my ($self, $variation_bed_writer, $genome) = @_;
+  foreach my $bed_file_path ($variation_bed_writer->variants_path($genome->genome_id())->children) {
+    print "Generating big bed from " . $bed_file_path->basename;
+    my $indexer = Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::IndexBed->variants($genome);
+    $indexer->index($bed_file_path);
+    $bed_file_path->remove;
+  }
+}
+
+
+
+
+
+sub get_vcf_files {
+
+  my ($self, $genome) = @_;
+  my $genome_type = $genome->type(); 
+  my $species = $genome->species();
+  my $ensembl_release = $self->param('ENS_VERSION');
+  my $ensembl_genomes_release = $self->param('EG_VERSION');
+  my $base_ftp_dir = '/nfs/production/panda/ensembl/production/ensemblftp/';
```

Comment:
Paths shouldn't be hard-coded in modules - needs to go in the pipe config and be passed in.

Reply:
ok i will add to  pipeline config

Reply:
It doesn't prevent the default path to be in the "default" method though.

Reply:
removed this line and added to default params in latest commits